trezor-crypto/crypto/tests/test_check.c

```diff
@@ -73,2048 +81,3771 @@
 #define DECRED_VERSION_PRIVATE 0x02fda4e8
 
 const uint8_t *fromhex(const char *str) {
-	static uint8_t buf[FROMHEX_MAXLEN];
-	size_t len = strlen(str) / 2;
-	if (len > FROMHEX_MAXLEN) len = FROMHEX_MAXLEN;
-	for (size_t i = 0; i < len; i++) {
-		uint8_t c = 0;
-		if (str[i * 2] >= '0' && str[i*2] <= '9') c += (str[i * 2] - '0') << 4;
-		if ((str[i * 2] & ~0x20) >= 'A' && (str[i*2] & ~0x20) <= 'F') c += (10 + (str[i * 2] & ~0x20) - 'A') << 4;
-		if (str[i * 2 + 1] >= '0' && str[i * 2 + 1] <= '9') c += (str[i * 2 + 1] - '0');
-		if ((str[i * 2 + 1] & ~0x20) >= 'A' && (str[i * 2 + 1] & ~0x20) <= 'F') c += (10 + (str[i * 2 + 1] & ~0x20) - 'A');
-		buf[i] = c;
-	}
-	return buf;
+  static uint8_t buf[FROMHEX_MAXLEN];
+  size_t len = strlen(str) / 2;
+  if (len > FROMHEX_MAXLEN) len = FROMHEX_MAXLEN;
+  for (size_t i = 0; i < len; i++) {
+    uint8_t c = 0;
+    if (str[i * 2] >= '0' && str[i * 2] <= '9') c += (str[i * 2] - '0') << 4;
+    if ((str[i * 2] & ~0x20) >= 'A' && (str[i * 2] & ~0x20) <= 'F')
+      c += (10 + (str[i * 2] & ~0x20) - 'A') << 4;
+    if (str[i * 2 + 1] >= '0' && str[i * 2 + 1] <= '9')
+      c += (str[i * 2 + 1] - '0');
+    if ((str[i * 2 + 1] & ~0x20) >= 'A' && (str[i * 2 + 1] & ~0x20) <= 'F')
+      c += (10 + (str[i * 2 + 1] & ~0x20) - 'A');
+    buf[i] = c;
+  }
+  return buf;
 }
 
 void nem_private_key(const char *reversed_hex, ed25519_secret_key private_key) {
-	const uint8_t *reversed_key = fromhex(reversed_hex);
-	for (size_t j = 0; j < sizeof(ed25519_secret_key); j++) {
-		private_key[j] = reversed_key[sizeof(ed25519_secret_key) - j - 1];
-	}
+  const uint8_t *reversed_key = fromhex(reversed_hex);
+  for (size_t j = 0; j < sizeof(ed25519_secret_key); j++) {
+    private_key[j] = reversed_key[sizeof(ed25519_secret_key) - j - 1];
+  }
 }
 
-START_TEST(test_bignum_read_be)
-{
-	bignum256 a;
-	uint8_t input[32];
+START_TEST(test_bignum_read_be) {
+  bignum256 a;
+  uint8_t input[32];
 
-	memcpy(input, fromhex("c55ece858b0ddd5263f96810fe14437cd3b5e1fbd7c6a2ec1e031f05e86d8bd5"), 32);
+  memcpy(
+      input,
+      fromhex(
+          "c55ece858b0ddd5263f96810fe14437cd3b5e1fbd7c6a2ec1e031f05e86d8bd5"),
+      32);
 
-	bn_read_be(input, &a);
+  bn_read_be(input, &a);
 
-	bignum256 b = { { 0x286d8bd5, 0x380c7c17, 0x3c6a2ec1, 0x2d787ef5, 0x14437cd3, 0x25a043f8, 0x1dd5263f, 0x33a162c3, 0x0000c55e } };
+  bignum256 b = {{0x086d8bd5, 0x1018f82f, 0x11a8bb07, 0x0bc3f7af, 0x0437cd3b,
+                  0x14087f0a, 0x15498fe5, 0x10b161bb, 0xc55ece}};
 
-	for (int i = 0; i < 9; i++) {
-		ck_assert_int_eq(a.val[i], b.val[i]);
-	}
+  for (int i = 0; i < 9; i++) {
+    ck_assert_int_eq(a.val[i], b.val[i]);
+  }
 }
 END_TEST
 
-START_TEST(test_bignum_write_be)
-{
-	bignum256 a = { { 0x286d8bd5, 0x380c7c17, 0x3c6a2ec1, 0x2d787ef5, 0x14437cd3, 0x25a043f8, 0x1dd5263f, 0x33a162c3, 0x0000c55e } };
-	uint8_t tmp[32];
+START_TEST(test_bignum_write_be) {
+  bignum256 a = {{0x086d8bd5, 0x1018f82f, 0x11a8bb07, 0x0bc3f7af, 0x0437cd3b,
```

Comment:
Why differnt?

Reply:
I have just realized that the bignum internal representation got changed, see: 
https://github.com/trustwallet/wallet-core/pull/1294/files#diff-9047eb00d4368211abc67ece44317e2f1b35dbe09d2a0c03f3d2d79385f58bd3
We use trezor bignum only in a few places, and in a way that it is not affected by the change in internal representation, but I will do some more testing on this.